### PR TITLE
MULTISITE-21181: Drush command for automatic check of contrib modules 

### PR DESCRIPTION
**Pull request by zarabatana**

### CODE REVIEW

File: includes/drush/custom/toolkit_check_contrib.drush.inc [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  }
+}
+
+// Helper function to get the list of authorised modules.
+function getModulesList() {
+  // Get list of authorised modules.
+  $url = 'https://raw.githubusercontent.com/ec-europa/qa-tests/components/module_list.json';
+
+  $request_headers = array(
+    'Authorization: token ' . getenv('EC_GITHUB_TOKEN')
```
Comment (ghost) [inline includes/drush/custom/toolkit_check_contrib.drush.inc]: I would rename this to GITHUB_API_TOKEN which is the standard.

If the token is not set there is no use in running the command. Maybe then abort the command and notify the user of the missing token?

File: includes/drush/custom/toolkit_check_contrib.drush.inc [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+  $curl = curl_init($url);
+  curl_setopt($curl, CURLOPT_HTTPHEADER, $request_headers);
+  curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+  $result = curl_exec($curl);
+
+  // If request did not fail.
+  if ($result !== false) {
+    // Request was ok? check response code.
+    $statusCode = curl_getinfo($curl, CURLINFO_HTTP_CODE);
+    if ($statusCode == 200) {
```
Comment (ghost) [inline includes/drush/custom/toolkit_check_contrib.drush.inc:199]: If status is not 200 we should return an error message explaining why the curl failed.

**Review by ghost — CHANGES_REQUESTED** [review]

File: includes/drush/custom/toolkit_check_contrib.drush.inc (hunk 2 above)
Comment (jonhy81) [reply includes/drush/custom/toolkit_check_contrib.drush.inc:199]: Done

File: includes/drush/custom/toolkit_check_contrib.drush.inc (hunk 1 above)
Comment (jonhy81) [reply includes/drush/custom/toolkit_check_contrib.drush.inc]: Done, just replaced. Need to discuss a possible fallback later.

Comment (jonhy81) [reply includes/drush/custom/toolkit_check_contrib.drush.inc]: https://webgate.ec.europa.eu/CITnet/jira/browse/MULTISITE-21456

**Review by ghost — APPROVED** [review]

**Pull request merged**